### PR TITLE
[GPU] Support implicit crop in input transposition.

### PR DESCRIPTION
**Pull request by jade-cho**
+ Make the crop in front of quantize implicit by changing output format to bfyx.
+ Choose onednn jit:ir as a kernel of first convolution, if feature size of input tensor is less than or equal to 4.

Ticket
- *82742*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 1]
```diff
@@ -1597,6 +1602,19 @@ format layout_optimizer::get_preferred_format(program_node& node) {
...
+
+        // In case of input -> crop -> quantize -> concat
+        if (use_onednn_impls && expected == format::any && node.get_dependency(0).is_type<crop>()
+            && (node.get_users().size() == 1 && node.get_users().front()->is_type<concatenation>())
+            && (layout.batch() < 4 && layout.feature() < 4)) {
+            if (layout.get_rank() == 6) {
+                expected = format::bfwzyx;
+            } else if (layout.get_rank() == 5) {
+                expected = format::bfzyx;
+            } else if (layout.get_rank() == 4) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1614]: I think you can use get_default_format

Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1610]: Oh~ I got it.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp [hunk 2]
```diff
@@ -353,7 +353,7 @@ void prepare_buffer_fusing::run(program& p) {
                 // do not optimize crop if paddings are not properly aligned
                 for (auto& usr : node.get_users()) {
                     auto usr_layout = usr->get_output_layout();
-                    if (usr_layout.format == format::b_fs_yx_fsv16 &&
+                    if (!usr->is_type<quantize>() && usr_layout.format == format::b_fs_yx_fsv16 &&
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:356]: Why do we need special treatment for quantize?

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 3]
```diff
@@ -23,6 +23,7 @@
 #include "mvn_inst.h"
 #include "depth_to_space_inst.h"
 #include "region_yolo_inst.h"
+#include "crop_inst.h"
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:26]: Is it necessary?

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 4]
```diff
@@ -1597,6 +1598,13 @@ format layout_optimizer::get_preferred_format(program_node& node) {
                     expected = expected_conv_fmt;
             }
         }
+
+        // In case of input -> ... -> quantize -> concat
+        if (expected == format::any
+            && (node.get_users().size() == 1 && node.get_users().front()->is_type<concatenation>())
+            && (layout.batch() < 4 && layout.feature() < 4)) {
+                expected = format::get_default_format(layout.get_rank(), false, false);
+        }
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1607]: How do you think about adding a test case for this pattern?

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp (hunk 3 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:26]: I will remove it.

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp (hunk 4 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1607]: I've found only one model has this pattern. So, I think it is not necessary.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp (hunk 2 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:356]: quantize follows the format of input. But, at this time, passing prepare buffer fusing, quantize format is bfyx. Therefore, to ignore format checking, this condition is added.

Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp]: I removed this condition. it doesn't need anymore.

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp (hunk 4 above)
Comment (jade-cho) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1606]: I've added a test case for this.

**Review by hyunback — APPROVED** [review]

**Pull request merged**